l1-contracts/src/core/libraries/crypto/SampleLib.sol

```diff
@@ -44,7 +44,7 @@ library SampleLib {
     uint256 upperLimit = _indexCount - 1;
 
     for (uint256 index = 0; index < _committeeSize; index++) {
-      uint256 sampledIndex = computeSampleIndex(index, upperLimit, _seed);
+      uint256 sampledIndex = computeSampleIndex(index, upperLimit + 1, _seed);
```

Comment:
Good spot, think we could probably do something with the naming to make it clearer.